base/linalg/symmetric.jl

```diff
@@ -107,6 +107,21 @@ end
     r
 end
 
+function setindex!(A::Symmetric, v, i::Integer, j::Integer)
+    i == j || throw(ArgumentError("Cannot set a non-diagonal index in a symmetric matrix"))
```

Comment:
This change doesn't make sense to me. Why shouldn't you be able to set an off diagonal element?